paddle/fluid/framework/transfer_scope_cache.cc

```diff
@@ -17,14 +17,31 @@
 namespace paddle {
 namespace framework {
 
+static std::unordered_map<size_t, Scope*>* static_transfer_data_cache = nullptr;
+static std::unordered_set<Scope*>* static_transfer_scope_cache = nullptr;
+
 std::unordered_map<size_t, Scope*>& global_transfer_data_cache() {
-  thread_local auto* x = new std::unordered_map<size_t, Scope*>;
-  return *x;
+  // if get_cur_thread_id() == -1, means not use thread local method to do cache
```

Comment:
Add more comments `platform::get_cur_thread_id() == -1` is only used in MKLDNN.
OR: 
```
#ifdef PADDLE_WITH_CUDA
    thread_local auto* x = new std::unordered_map<size_t, Scope*>;
    return *x;
#else
#ifdef PADDLE_WITH_MKLDNN
   if (platform::get_cur_thread_id() == -1) {
      if (!static_transfer_data_cache)
         static_transfer_data_cache = new std::unordered_map<size_t, Scope*>;
   }
#endif
#endif
```
?

Reply:
good idea, to be updated.

Reply:
it is not special mkldnn issue, but this solution only works with mkldnn api.